pkg/graph/resolvers/model_plan.go

```diff
@@ -247,6 +248,24 @@ func ModelPlanGetByIDLOADER(ctx context.Context, id uuid.UUID) (*models.ModelPla
 	return result.(*models.ModelPlan), nil
 }
 
+// ModelPlanTrackingDateGetByIDLOADER implements resolver logic to get Model Plan
+// Tracking Date by a model plan ID using a data loader
+func ModelPlanTrackingDateGetByIDLOADER(ctx context.Context, id uuid.UUID) (*time.Time, error) {
+	allLoaders := loaders.Loaders(ctx)
+	planTrackingDateLoader := allLoaders.ModelPlanTrackingDateLoader
+	key := loaders.NewKeyArgs()
+	key.Args["id"] = id
+
+	thunk := planTrackingDateLoader.Loader.Load(ctx, key)
+	result, err := thunk()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result.(*time.Time), nil
```

Comment:
We do this in a couple places, but this I've recently seen that this can make a panic. Could you please do the type assertion a line before to validate that the result before returning?

Something like
```suggestion
	timeRes, isTime := result.(*time.Time)
	if !isTime {
		return nil, fmt.Errorf("the result was not *time.Time as expected for the solution section tracking date, it was %T", result)
	}
	return timeRes, nil
```